libbeat/processors/dissect/field.go

```diff
@@ -90,8 +123,56 @@ type normalField struct {
 	baseField
 }
 
+// strToInt is a helper to interpret a string as either base 10 or base 16.
+func strToInt(s string, bitSize int) (int64, error) {
+	base := 10
+	if strings.HasPrefix(s, "0x") || strings.HasPrefix(s, "0X") {
+		// strconv.ParseInt will accept the '0x' or '0X` prefix only when base is 0.
+		base = 0
+	}
+	return strconv.ParseInt(s, base, bitSize)
+}
+
+func transformType(typ dataType, value string) (interface{}, error) {
+	switch typ {
+	case String:
+		return fmt.Sprintf("%v", value), nil
```

Comment:
I think this conversion is not needed, the value is already a string.
```suggestion
		return value, nil
```

Reply:
Removed the redundant conversion.